autoBOTLib/features/features_reading_comperhension.py

```diff
@@ -0,0 +1,252 @@
+    pl = len(long_words) / lengthW * 100  # percentage long words
+    GFI = 0.4 * ((lengthW / lengthS) + pl)
+    return GFI
+
+
+def fre(text):
+    # Flesch Reading Ease
+    word_tokens = regexp_tokenize(text, r'\w+')
+    lengthW = len(word_tokens)
+    sents = sent_tokenize(text)
+    lengthS = len(sents)
+
+    # Avoid division by zero if no sentences or no words
+    if lengthS == 0 or lengthW == 0:
+        return 0
+
+    ts = 0  # total syllables
+    for word in word_tokens:
+        ts += sylco(word)
+
+    FRE = 206.835 - 1.015 * (lengthW / lengthS) - 84.6 * (ts / lengthW)
+    return FRE
+
+
+def fkgl(text):
+    # Flesch–Kincaid Grade Level
+    word_tokens = regexp_tokenize(text, r'\w+')
+    lengthW = len(word_tokens)
+    sents = sent_tokenize(text)
+    lengthS = len(sents)
+
+    # Avoid division by zero if no sentences or no words
+    if lengthS == 0 or lengthW == 0:
+        return 0
+
+    ts = 0  # total syllables
+    for word in word_tokens:
+        ts += sylco(word)
+
+    FKGL = 0.39 * (lengthW / lengthS) + 11.8 * (ts / lengthW) - 15.59
+    return FKGL
+
+
+def dcrf(text):
+    # Dale–Chall Readability Formula
+    word_tokens = regexp_tokenize(text, r'\w+')
+    lengthW = len(word_tokens)
+    sents = sent_tokenize(text)
+    lengthS = len(sents)
+
+    # Avoid division by zero
+    if lengthS == 0 or lengthW == 0:
+        return 0
+
+    long_words = [w for w in word_tokens if len(w) > 7]
+    pl = len(long_words) / lengthW * 100  # percentage of long words
+
+    DCRF = 0.1579 * pl + 0.0496 * (lengthW / lengthS)
+    return DCRF
+
+
+def ari(text):
+    # Automated Readability Index
+    word_tokens = regexp_tokenize(text, r'\w+')
+    lengthW = len(word_tokens)
+    sents = sent_tokenize(text)
+    lengthS = len(sents)
+    lengthCH = len(text)
+
+    # Avoid division by zero
+    if lengthW == 0 or lengthS == 0:
+        return 0
+
+    ARI = 4.71 * (lengthCH / lengthW) + 0.5 * (lengthW / lengthS) - 21.43
+    return ARI
+
+
+def smog(text):
+    # SMOG Index
+    word_tokens = regexp_tokenize(text, r'\w+')
+    sents = sent_tokenize(text)
+    lengthS = len(sents)
+
+    if lengthS == 0:
+        return 0
+
+    tps = 0  # total words with more than 2 syllables
+    for word in word_tokens:
+        if sylco(word) > 2:
+            tps += 1
+
+    SMOG = 1.043 * math.sqrt(tps * (30 / lengthS)) + 3.1291
+    return SMOG
+
+
+def sent_len(text):
+    sents = nltk.sent_tokenize(text)
+    if not sents:
+        return 0
+    # Calculate average sentence length (in words)
+    text_lens = [len(nltk.word_tokenize(sent)) for sent in sents]
+    if len(text_lens) == 0:
+        return 0
+    return sum(text_lens) / len(text_lens)
+
+
+def ttr(text):
+    # Type-Token Ratio (vocabulary diversity)
+    words = [word for sent in nltk.sent_tokenize(text) for word in nltk.word_tokenize(sent)]
+    if len(words) == 0:
+        return 0
+    return len(set(words)) / len(words)
+
+
+class ComperhensionFeatures:
```

Comment:
WordComprehension?

Reply:
Since it's reading comperhension on the whole text, I suggest we keep the current name.

Reply:
DocumentComprehension?